app/bundles/CoreBundle/Twig/Extension/LanguageExtension.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Mautic\CoreBundle\Twig\Extension;
+
+use Mautic\UserBundle\Entity\User;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\Intl\Languages;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+class LanguageExtension extends AbstractExtension
+{
+    public function __construct(private Security $security)
+    {
+    }
+
+    public function getFilters()
+    {
+        return [
+            new TwigFilter('language_name', [$this, 'getLanguageName']),
+        ];
+    }
+
+    /**
+     * Returns the language name for the given language code.
+     *
+     * @param string      $code          The language code (e.g., 'en', 'fr', etc.)
+     * @param string|null $displayLocale The locale used to display the language name (defaults to user's locale)
+     *
+     * @return string The language name
+     */
+    public function getLanguageName(string $code, ?string $displayLocale = null): string
+    {
+        if (null === $displayLocale) {
+            $displayLocale = $this->getCurrentUserLocale();
+        }
+
+        try {
+            return Languages::getName($code, $displayLocale) ?: $code;
+        } catch (\Exception) {
```

Comment:
Let's stop catching general exceptions as it masks actual errors that are then missing in the logs and it makes them very hard to investigate.
```suggestion
        } catch (MissingResourceException) {
```
+ add the use statement for this exception.